src/actions/RemovePaintComponent.java

```diff
@@ -45,12 +49,16 @@ public void redoAction() {
 					comp.remove(panel);
 			}
 		});
+
 		panel.repaint();
 	}
 
 	@Override
 	public String locationString() {
 		// TODO Auto-generated method stub
-		return ActionsMenuBarTitles.Data().Remove().toString();	}
-
+		if(panel.getSelectTool() instanceof SmartTool)
```

Comment:
This approach (using instanceof) is dangerous, Please Comment on these two lines on the reason for selection.

Instanceof makes code unmaintanable.